docs/custom-directive.md

```diff
@@ -127,3 +127,92 @@ app.register(mercurius, {
 ## Example
 
 We have a runnable example on "example/custom-directive.js"
+
+# Federation and Custom Directives
+
+If we are using Federation, we must follow a different approach to create custom directives.
```

Comment:
It would be helpful to explain why our existing approach won't work if we are using federation. And also to introduce readers to what we will be doing next (i.e. demonstrate by creating a new `@upper` custom directive)

Reply:
I added more info and reviewed this part.
Let me know if it's better now, please.